src/Symfony/Bundle/FrameworkBundle/DataCollector/MessengerDataCollector.php

```diff
@@ -77,7 +84,7 @@ public function handle($message, callable $next)
             );
         }
 
-        $this->data[] = $debugRepresentation;
+        $this->data['messages'][] = $debugRepresentation;
```

Comment:
`collect` must ensure that `$this->data['messages']` is always an array, even when no message was handled there, to make `getMessages` work without notice after `collect` has been called.

Reply:
Thank you

Reply:
Doesn't it? 
Cant I trust that `$this->data` always to be null or an array?

https://3v4l.org/tubTL

Reply:
Actually, you're right, it won't "throw" a PHP notice.